lib/smooth/application/display/DisplaySpi.cpp

```diff
@@ -59,87 +58,146 @@ namespace smooth::application::display
+        std::this_thread::sleep_for(std::chrono::milliseconds(delay_time));
+
+        return res;
     }
 
     // Send initialize sequence of commands and data to display
-    bool ILI9341::send_init_cmds()
+    bool DisplaySpi::send_init_cmds(const display_init_cmd_t* init_cmds, size_t length)
     {
         bool res = true;
 
-        uint16_t seq_num = 0;
-
-        while (ili_init_cmds.at(seq_num).databytes != 0xff)
+        for (size_t index = 0; index < length; index++)
         {
-            // check if delay is needed; send only command then delay
-            if (ili_init_cmds.at(seq_num).databytes & 0x80)
+            // send command
+            res &= send_cmd(init_cmds[index].cmd);
+
+            // send data
+            uint8_t data_length = init_cmds[index].length;
+
+            if (data_length >= 1 && data_length < 0x80)
             {
-                res &= lcd_wr_cmd(ili_init_cmds.at(seq_num).cmd);
-                std::this_thread::sleep_for(std::chrono::milliseconds(100));
+                res &= send_data(init_cmds[index].data, data_length);
             }
-            else
+
+            // check to see if we have to delay before advancing to next sequence in ints_cmds
+            if (data_length >= 0x80)
             {
-                res &= lcd_wr_cmd(ili_init_cmds.at(seq_num).cmd);
-                res &= lcd_wr_data(ili_init_cmds.at(seq_num).data, ili_init_cmds.at(seq_num).databytes & 0x1F);
+                uint8_t delay_data_length = static_cast<uint8_t>(data_length & 0x7F);
+                size_t delay_time = init_cmds[index].data[delay_data_length];
+
+                // check delay_data_length to see if there is data to send before delaying
+                if (delay_data_length != 0)
+                {
+                    res &= send_data(init_cmds[index].data, delay_data_length);
+                }
+
+                // delay value of 255 is really 500
+                delay_time = delay_time == 255 ? 500 : delay_time;
+                std::this_thread::sleep_for(std::chrono::milliseconds(delay_time));
             }
-
-            // advance to next sequence in ili_ints_cmds
-            seq_num++;
         }
 
         return res;
     }
 
-    // Send a command to the LCD. Uses spi_device_polling_write, which waits
-    // until the transfer is complete.
+    // Send muliple commands to the display.
+    // Uses spi_device_polling_write, which waits until the transfer is complete.
+    bool DisplaySpi::send_cmds(const uint8_t* cmds, size_t length)
+    {
+        // reset current transaction counter
+        current_transaction = 0;
+
+        // setup pretrans data/command pin state
+        dc_pretrans_pin_states[0] = PIN_LOW;
+        cs_pretrans_pin_states[0] = PIN_LOW;
+        cs_posttrans_pin_states[0] = PIN_HIGH;
+
+        std::lock_guard<std::mutex> lock(get_guard());
+        spi_transaction_t trans;
+        std::memset(&trans, 0, sizeof(trans));  //Zero out the transaction
```

Comment:
Use zero-intialization instead of the memset: `spi_transaction_t trans{};`

https://en.cppreference.com/w/cpp/language/zero_initialization